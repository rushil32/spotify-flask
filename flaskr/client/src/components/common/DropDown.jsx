import React from 'react';
import PropTypes from 'prop-types';

const DropDown = ({ list, handleClick, width }) => {
  return (
    <div className="form-dropdown" style={{ width: `${width}px` }}>
      <ul>
        {list.map((item, index) => (
          <li key={index} onClick={() => handleClick(item)}>
            <span>{item.header}</span>
            <span>{item.subheader}</span>
          </li>
        ))}
      </ul>
    </div>
  );
};

DropDown.defaultProps = {
  width: '100px',
  handleClick: () => {},
  list: [],
};

DropDown.propTypes = {
  width: PropTypes.string,
  handleClick: PropTypes.func,
  list: PropTypes.array,
};

export default DropDown;
