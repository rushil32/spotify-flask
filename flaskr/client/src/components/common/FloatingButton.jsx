import React from "react";

const FloatingButton = ({ handleClick, tooltip, icon }) => (
  <button
    className="btn btn-float animated flipInX"
    data-toggle="tooltip"
    data-placement="top"
    data-offset="0, 5"
    title={tooltip}
    type="button"
    onClick={handleClick}
  >
    <i className="material-icons">{icon}</i>
  </button>
);

export default FloatingButton;
