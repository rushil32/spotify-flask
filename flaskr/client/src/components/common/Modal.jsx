import React from 'react';
import PropTypes from 'prop-types';

const Modal = ({ name, header, icon, children }) => (
  <div className="modal fade" id={name} tabIndex="-1" role="dialog">
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <span className="modal-icon">
            <i className="material-icons">{icon}</i>
          </span>
          <h5 className="modal-title">{header}</h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">{children}</div>
      </div>
    </div>
  </div>
);

Modal.defaultProps = {
  name: '',
  header: '',
  icon: 'plus',
  children: <div />,
};

Modal.propTypes = {
  name: PropTypes.string,
  header: PropTypes.string,
  icon: PropTypes.string,
  children: PropTypes.object
};

export default Modal;
