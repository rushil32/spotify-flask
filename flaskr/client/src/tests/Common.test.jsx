import React from 'react';
import renderer from 'react-test-renderer';

import data from './mock-data';
import FloatingButton from '../components/common/FloatingButton';
import DropDown from '../components/common/DropDown';
import Modal from '../components/common/Modal';

it('renders the floating button correctly', () => {
  const tree = renderer
    .create(<FloatingButton
      icon="group_add"
      tooltip="Start Room"
      handleClick={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the dropdown correctly', () => {
  const tree = renderer
    .create(<DropDown
      list={data.dropdown}
      tooltip="Tooltip test"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the modal correctly', () => {
  const tree = renderer
    .create(<Modal />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
