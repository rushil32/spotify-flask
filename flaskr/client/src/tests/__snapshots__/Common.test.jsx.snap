// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dropdown correctly 1`] = `
<div
  className="form-dropdown"
  style={
    Object {
      "width": "100pxpx",
    }
  }
>
  <ul>
    <li
      onClick={[Function]}
    >
      <span>
        First option
      </span>
      <span>
        The first option!
      </span>
    </li>
    <li
      onClick={[Function]}
    >
      <span>
        Second option
      </span>
      <span>
        The second option!
      </span>
    </li>
    <li
      onClick={[Function]}
    >
      <span>
        Third option
      </span>
      <span>
        The third option!
      </span>
    </li>
  </ul>
</div>
`;

exports[`renders the floating button correctly 1`] = `
<button
  className="btn btn-float animated flipInX"
  data-offset="0, 5"
  data-placement="top"
  data-toggle="tooltip"
  onClick={[Function]}
  title="Start Room"
  type="button"
>
  <i
    className="material-icons"
  >
    group_add
  </i>
</button>
`;

exports[`renders the modal correctly 1`] = `
<div
  className="modal fade"
  id=""
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <span
          className="modal-icon"
        >
          <i
            className="material-icons"
          >
            plus
          </i>
        </span>
        <h5
          className="modal-title"
        >
          
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
